pkg/apis/kubernetes/v1/kubeconfig_helper.go

```diff
@@ -29,7 +29,7 @@ func GetKubeConfig(ctx context.Context, a api.API, clusterID string) (string, er
 			return "", fmt.Errorf("failed to get cluster: %w", err)
 		}
 
-		if cluster.KubeConfig != nil {
+		if pointer.StringVal(cluster.KubeConfig) != "" {
```

Comment:
Don't know for sure if this was simply a race condition (probably not) or if the behavior of uninitialized fields has changed. According to previous integrations tests this has been working.